snuba/web/split.py

```diff
@@ -57,40 +90,53 @@ def time_split(dataset, request: Request, *args, **kwargs):
         into smaller increments, and start with the last one, so that we can potentially
         avoid querying the entire range.
         """
+        if not self.__can_execute(request):
+            return None
+
         date_align, split_step = state.get_configs(
             [("date_align_seconds", 1), ("split_step", 3600)]  # default 1 hour
         )
 
-        query_limit = request.query.get_limit()
-        limit = query_limit if query_limit is not None else 0
-        remaining_offset = request.query.get_offset()
-
-        to_date = util.parse_datetime(
-            request.extensions["timeseries"]["to_date"], date_align
+        # We already ensured the conditions below exist in the query during the
+        # can_execute call.
```

Comment:
Seems like it'd be better to just find them once then, now that there is no split between `execute` and `__can_execute` from the caller's perspective.

Reply:
Fair point, I forgot about that.